browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc

```diff
@@ -52,6 +56,53 @@ constexpr char kURLManagePremium[] = "https://account.brave.com/";
 
 namespace ai_chat {
 
+namespace {
+
+#if !BUILDFLAG(IS_ANDROID)
+// Invokes a callback when the WebContents has finished loading. Note: If the
+// WebContents is destroyed before loading is completed, the callback will not
+// be invoked.
+// The lifetime of this class is tied to the WebContents it is observing - it
+// will be destroyed when |WebContentsDestroyed| is called, or when the
+// Navigation finishes, whichever happens first.
+class WaitForCommit : public content::WebContentsObserver {
+ public:
+  WaitForCommit(
+      content::WebContents* contents,
+      base::OnceCallback<void(content::WebContents* contents)> on_loaded)
+      : WebContentsObserver(contents), on_loaded_(std::move(on_loaded)) {}
+  ~WaitForCommit() override = default;
+
+  void DidFinishNavigation(content::NavigationHandle* handle) override {
```

Comment:
```
// Listening to PrimaryPageChanged should be preferred to listening to
  // DidFinishNavigation and checking IsInPrimaryMainFrame, !IsSameDocument, and
  // HasCommitted.
```

Reply:
or maybe this doesn't apply here because it was `NeedsReload`?

Reply:
is it sufficient that the navigation has committed or do you need to wait until the page load completes?

Reply:
The primary page won't be changed. Waiting for the navigation to finish is sufficient. At this point, we just want to make sure the tab isn't suspended